### PR TITLE
Introduce network module, add picasso to hilt

### PR DESCRIPTION
**Pull request by naivekook**
####  Description
Took the liberty of refactoring network layer onto dagger approach.
Now hilt is responsible for creation of all APIs with http client and picasso.

#### Related issues
#3880 

#### Related PRs
none

#### Screenshots
none

### CODE REVIEW

**Comment by teolemon** [comment]
```
> Task :app:compileOffPlaystoreScreenshotsKotlin FAILED
e: /home/runner/work/openfoodfacts-androidapp/openfoodfacts-androidapp/app/src/playstore/java/openfoodfacts/github/scrachx/openfood/features/scan/ContinuousScanActivity.kt: (375, 102): No value passed for parameter 'picasso'

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:compileOffPlaystoreScreenshotsKotlin'.
> Compilation error. See log for more details
```

**Comment by naivekook** [comment]
ooh, we have two `ContinuousScanActivity`, not cool :(

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/hilt/ApiModule.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+@InstallIn(SingletonComponent::class)
+object ApiModule {
+
+    @Provides
+    @Singleton
+    fun provideProductsAPI(retrofit: Retrofit): ProductsAPI = retrofit.create(ProductsAPI::class.java)
+
+    @Provides
+    @Singleton
+    fun provideWikiDataAPI(retrofit: Retrofit): WikidataAPI = retrofit.create(WikidataAPI::class.java)
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/hilt/ApiModule.kt]: Wikidata does not use the same base url, and same for robotoff I think

Comment (naivekook) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/hilt/ApiModule.kt]: oh right, fixed

**Comment by VaiTon** [comment]
Could you please test the app in emu?

**Comment by teolemon** [comment]
There are some conflicts to solve.

**Comment by naivekook** [comment]
Conflicts are solved, did small smoke test, looks like everything is good

**Pull request merged**